options/db_options.cc

```diff
@@ -1073,18 +1073,22 @@ bool MutableDBOptions::IsNowOffPeak(SystemClock* clock) const {
   int64_t now;
   if (clock->GetCurrentTime(&now).ok()) {
     constexpr int kSecondsPerDay = 86400;
-    int since_midnight_seconds = static_cast<int>(now % kSecondsPerDay);
+    constexpr int kSecondsPerMinute = 60;
+    int seconds_since_midnight_to_nearest_minute =
+        (static_cast<int>(now % kSecondsPerDay) / kSecondsPerMinute) *
+        kSecondsPerMinute;
     int start_time = 0, end_time = 0;
     assert(
         TryParseTimeRangeString(daily_offpeak_time_utc, start_time, end_time));
```

Comment:
Code inside the assert will not be included in release build. You need something like `bool success = ...; assert(success); (void)success;` to avoid unused variable warning in release build.